pkg/controller/atlasproject/custom_roles.go

```diff
@@ -31,12 +31,26 @@ func hasSkippedCustomRoles(atlasProject *akov2.AtlasProject) (bool, error) {
 			return false, fmt.Errorf("failed to parse last skipped configuration: %w", err)
 		}
 
-		return len(lastSkippedSpec.CustomRoles) == 0, nil
+		return len(lastSkippedSpec.CustomRoles) != 0, nil
 	}
 
 	return false, nil
 }
 
+func hasLastAppliedCustomRoles(atlasProject *akov2.AtlasProject) (bool, error) {
+	lastAppliedSpec := akov2.AtlasProjectSpec{}
+	lastAppliedSpecStr, ok := atlasProject.Annotations[customresource.AnnotationLastAppliedConfiguration]
+	if !ok {
+		return false, nil
+	}
+
+	if err := json.Unmarshal([]byte(lastAppliedSpecStr), &lastAppliedSpec); err != nil {
+		return false, fmt.Errorf("failed to parse last applied configuration: %w", err)
+	}
+
+	return len(lastAppliedSpec.CustomRoles) != 0, nil
```

Comment:
As many other sub-resources, it is a slice of items. This implementation is an all or nothing rule, as we allowed mixed modes, this doesn't allow different subsets of custom roles of being managed here and in a separate resource. Use case: Customer has a project with custom roles and start creating new ones from the new resource while plan the migration.
cc @igor-karpukhin @s-urbaniak

Reply:
Sorry, I did not get that. Do you mean this change broke the migration process for custom roles?